rasa/shared/core/domain.py

```diff
@@ -251,64 +229,24 @@ def from_directory(cls, path: Text) -> "Domain":
                     other_dict = rasa.shared.utils.io.read_yaml(
                         rasa.shared.utils.io.read_file(full_path)
                     )
-                    domain_dict = Domain.merge_domain_dicts(
-                        cls, domain_dict, other_dict
-                    )
+                    domain_dict = Domain.merge(domain_dict, other_dict, is_dir=True)
+
         domain = Domain.from_dict(domain_dict)
         return domain
 
-    def merge(self, domain: Optional["Domain"], override: bool = False) -> "Domain":
-        """Merge this domain with another one, combining their attributes.
-
-        List attributes like ``intents`` and ``actions`` will be deduped
-        and merged. Single attributes will be taken from `self` unless
-        override is `True`, in which case they are taken from `domain`.
-        """
-        if not domain or domain.is_empty():
-            return self
-
-        if self.is_empty():
-            return domain
-
-        domain_dict = domain.as_dict()
-        combined = self.as_dict()
-
-        if override:
-            config = domain_dict["config"]
-            for key, val in config.items():
-                combined["config"][key] = val
-
-        if override or self.session_config == SessionConfig.default():
-            combined[SESSION_CONFIG_KEY] = domain_dict[SESSION_CONFIG_KEY]
-
-        for key in [KEY_INTENTS, KEY_ENTITIES]:
-            if combined[key] or domain_dict[key]:
-                combined[key] = self.merge_lists_of_dicts(
-                    combined[key], domain_dict[key], override
-                )
-        # remove existing forms from new actions
-        for form in combined[KEY_FORMS]:
-            if form in domain_dict[KEY_ACTIONS]:
-                domain_dict[KEY_ACTIONS].remove(form)
-
-        for key in [KEY_ACTIONS, KEY_E2E_ACTIONS]:
-            combined[key] = self.merge_lists(combined[key], domain_dict[key])
-
-        for key in [KEY_FORMS, KEY_RESPONSES, KEY_SLOTS]:
-            combined[key] = self.merge_dicts(combined[key], domain_dict[key], override)
-
-        return self.__class__.from_dict(combined)
-
-    def merge_domain_dicts(
-        self, domain1: Dict, domain2: Dict, override: bool = False
+    @classmethod
```

Comment:
I like this `classmethod` as the core merging function - I wonder if it would also be useful to have a lightweight `merge` instance method that just calls this. That way we could still do stuff like `domain.merge(other)`?